packages/app/src/client/views/ring-health/RingTable.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2021 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from "react";
+
+import axios from "axios";
+import {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody
+} from "@material-ui/core";
+import { Card } from "client/components/Card";
+import { format } from "date-fns";
+
+type Shard = {
+  id: string;
+  state: string;
+  timestamp: string;
+  zone: string;
+  address: string;
+  tokens: number[];
+  registered_timestamp: string;
+};
+
+type Payload = {
+  shards: Array<Shard>;
+  now: Date;
+};
+
+type Props = {
+  ringEndpoint: string;
+};
+
+const RingTable = ({ ringEndpoint }: Props) => {
+  const [shards, setShards] = useState<Array<Shard>>();
+  useEffect(() => {
+    async function fetchShards() {
+      const response = await axios.get<Payload>(ringEndpoint);
+      setShards(response.data.shards);
+    }
+    fetchShards();
+  }, [ringEndpoint]);
+
+  return (
+    <TableContainer component={Card}>
+      <Table aria-label="tenants">
+        <TableHead>
+          <TableRow>
+            <TableCell>ID</TableCell>
+            <TableCell>State</TableCell>
+            <TableCell>last Heartbeat</TableCell>
+            <TableCell>Zone</TableCell>
+            <TableCell>Address</TableCell>
+            <TableCell>Tokens</TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {shards?.map(shard => (
```

Comment:
It'd probably be better to show some spinner. Do we already have something like this - to keep things consistent?

Reply:
We use the <Skeleton /> component for loading. In this case, instead of showing the table during loading, you can show 4 horizontal skeletons that looks like the following: https://material-ui.com/components/skeleton/#animations